tests/unit/test_utils.py

```diff
@@ -524,3 +525,10 @@ def test_manylinux1_check_glibc_version(self):
                 else:
                     # Didn't find the warning we were expecting
                     assert False
+
+
+class Test_user_cache_dir(object):
+
+    def test_return_path_as_str(self):
+        """Test that path is bytes on Python 2 and Unicode on Python 3."""
+        assert isinstance(user_cache_dir('test'), str)
```

Comment:
I think I would go with something like:

    @pytest.mark.skipif("sys.platform != 'win32'")
    @patch('pip.utils.appdirs._get_win_folder', _get_win_folder)
    def test_your_test(self, _get_win_folder):
        _get_win_folder.return_value = u"läppäri"
        assert isinstance(user_cache_dir('test'), str)
        # Test against regression #3463
        from pip import create_main_parser
        create_main_parser().print_help()  # This should not crash